src/wallet/spend.cpp

```diff
@@ -434,9 +434,17 @@ std::optional<SelectionResult> SelectCoins(const CWallet& wallet, const std::vec
              */
             preset_inputs.Insert(out, /*ancestors=*/ 0, /*descendants=*/ 0, /*positive_only=*/ false);
         }
+        if (preset_inputs.GetSelectionAmount() < nTargetValue) return std::nullopt;
```

Comment:
Wasn't this what we introduced `SelectionResult.Merge(…)` above for?

Reply:
Not sure what did you mean by this. This is the case when the selected inputs are not enough and we just return null value. There's no merging going on.

Reply:
Right, but if the preset inputs are less than the target, wouldn't we be supposed to select additional inputs and build a transaction? I thought that's what you introduced the merge for.